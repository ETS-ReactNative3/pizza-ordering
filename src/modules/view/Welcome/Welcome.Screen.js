import React, {Component} from 'react';
import {Image, View, Text, TouchableOpacity} from 'react-native';
import welcomeStyle from './Welcome.style';
import welcomeStrings from '../../resources/strings/Welcome.strings';

const localImage = require('../../resources/images/welcome03.jpg');

export default class Welcome extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <View style={welcomeStyle.container}>
        <Image style={welcomeStyle.image} source={localImage} />
        <View style={welcomeStyle.titleContainer}>
          <Text style={welcomeStyle.restaurantText}>
            {welcomeStrings.restaurant}
          </Text>
          <Text style={welcomeStyle.pizzaText}>{welcomeStrings.name}</Text>
          <TouchableOpacity
            onPress={() => this.props.navigation.navigate('MainDrawer')}>
            <View style={welcomeStyle.buttonContainer}>
              <Text style={welcomeStyle.buttonText}>
                {welcomeStrings.startOrder}
              </Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
