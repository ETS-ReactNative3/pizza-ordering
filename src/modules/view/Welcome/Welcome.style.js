import {StyleSheet} from 'react-native';
import colors from '../../resources/colors/Colors';
import dimensions from '../../resources/dimensions/Dimensions';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column-reverse',
  },
  image: {
    flex: 1,
    width: null,
    height: null,
    resizeMode: 'cover',
  },
  titleContainer: {
    flexDirection: 'column',
    position: 'absolute',
    margin: 32,
  },
  restaurantText: {
    fontSize: 40,
    color: colors.text,
    fontWeight: 'bold',
  },
  pizzaText: {
    fontSize: 25,
    color: colors.text,
  },
  buttonContainer: {
    flexDirection: 'row',
    backgroundColor: colors.mainRed,
    borderRadius: 20,
    justifyContent: 'center',
  },
  buttonText: {
    fontSize: 17,
    color: colors.text,
    fontWeight: 'bold',
    alignSelf: 'center',
    height: dimensions.startOrderBtn.height,
    lineHeight: dimensions.startOrderBtn.lineHeight,
    margin: 2,
  },
});
