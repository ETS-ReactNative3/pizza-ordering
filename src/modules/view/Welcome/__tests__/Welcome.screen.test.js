import React from 'react';
import {mount, shallow} from 'enzyme';

import Welcome from '../Welcome.Screen';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Welcome screen test', () => {
  const wrapper = shallow(<Welcome />);

  it('should render snapshot test for Welcome', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should navigate categories screen', function() {
    const props = {navigation: {navigate: jest.fn()}};
    const wrapper = mount(<Welcome {...props} />);
    const btnStartOrder = wrapper.findWhere(node => {
      return node.text() === 'START ORDER >' && typeof node.type() === 'string';
    });
    btnStartOrder.at(1).simulate('click');

    expect(props.navigation.navigate).toBeCalledWith('MainDrawer');
  });
});
