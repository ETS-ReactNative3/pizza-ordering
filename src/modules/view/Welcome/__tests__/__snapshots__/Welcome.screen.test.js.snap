// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome screen test should render snapshot test for Welcome 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column-reverse",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/modules/resources/images/welcome03.jpg",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": null,
        "resizeMode": "cover",
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 32,
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      Restaurant
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 25,
        }
      }
    >
      pyzza pizza
    </Text>
    <ForwardRef
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#E4283D",
            "borderRadius": 20,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "fontSize": 17,
              "fontWeight": "bold",
              "height": 50,
              "lineHeight": 50,
              "margin": 2,
            }
          }
        >
          START ORDER &gt;
        </Text>
      </View>
    </ForwardRef>
  </View>
</View>
`;
